### PR TITLE
Fix multiple tests with refactoring driver for duplicate class.

### PR DESCRIPTION
**Pull request by hernanmd**
Fix a couple of tests failing, reported in #16850, due to missing support to pre-configure refactorings for test purposes.
Also configured missing mocks for the preview presenter.
Fix also for #16825

### CODE REVIEW

**Comment by MarcusDenker** [comment]
Argh, I did merge the revert in the meantime so now this is a conflict.

**Comment by MarcusDenker** [comment]
I am doing a revert of the revert: https://github.com/pharo-project/pharo/pull/16854

**Comment by hernanmd** [comment]
> I am doing a revert of the revert: #16854

Thanks, I will check further fails and check better next time.
(and I forgot to mention I was working on this issue)

**Comment by MarcusDenker** [comment]
Revert fixed the conflicts, tests are now run

File: src/Refactoring-UI/ReDuplicateClassDriver.class.st [hunk 1]
```diff
@@ -99,6 +99,17 @@ ReDuplicateClassDriver >> runRefactoring [
 	self applyChanges 
 ]
 
+{ #category : 'as yet unclassified' }
```
Comment (MarcusDenker) [inline src/Refactoring-UI/ReDuplicateClassDriver.class.st]: this makes testNoUncategorizedMethods fail

**Review by MarcusDenker — CHANGES_REQUESTED** [review]
this adds one more failing testNoUncategorizedMethods

File: src/Refactoring-Core/RBCopyClassRefactoring.class.st [hunk 2]
```diff
@@ -150,41 +195,38 @@ ReCopyClassRefactoring >> copyVariables [
...
 { #category : 'transforming' }
 ReCopyClassRefactoring >> instanceMethods [
 
 	^ instanceMethods
+		ifNil: [ instanceMethods := aClass methods ]
 ]
 
-{ #category : 'accessing' }
-ReCopyClassRefactoring >> instanceMethods: anObject [
+||||||| 3926c07db6:src/Refactoring-Core/ReCopyClassRefactoring.class.st
```
Comment (MarcusDenker) [inline src/Refactoring-Core/RBCopyClassRefactoring.class.st]: this is now leading to load error

```
MetacelloNotification: Loaded -> Refactoring-Environment --- tonel:///builds/workspace/est_and_branch_Pipeline_PR-16853/64/src()
MetacelloNotification: Loaded -> Refactoring-Changes --- tonel:///builds/workspace/est_and_branch_Pipeline_PR-16853/64/src()
TonelParseError: Can't parse method body
```

**Comment by MarcusDenker** [comment]
Now load fails:

```
MetacelloNotification: Loaded -> Refactoring-Changes --- tonel:///builds/workspace/est_and_branch_Pipeline_PR-16853/64/src()
Error: Package Refactoring-Core depends on the following classes:
  ReCopyClassRefactoring
You must resolve these dependencies before you will be able to load these definitions: 
  ReCopyClassRefactoring>>#classMethods
  ReCopyClassRefactoring>>#classMethods:
  ReCopyClassRefactoring>>#copyClass
  ReCopyClassRefactoring>>#copyClass:withName:
  ReCopyClassRefactoring>>#copyMethods:of:in:
  ReCopyClassRefactoring>>#instanceMethods

```

**Comment by MarcusDenker** [comment]
Now all the test that the revert fixed are failing:

```

ReDuplicateClassDriverTest.testDuplicateClass	
 
 
 osx-64 / Tests-osx-64 / MacOSX64.Refactoring.Transformations.Tests.RBCopyPackageParametrizedTest.testCopyPackageAndChangesCopyReferences(#rbClass->RBCopyPackageRefactoring)	6.8 sec	1
 osx-64 / Tests-osx-64 / MacOSX64.Refactoring.Transformations.Tests.RBCopyPackageParametrizedTest.testCopyPackageWithParameters(#rbClass->RBCopyPackageRefactoring)
```

In addition, we have:

ReleaseTest>>testUndeclared


It might fix other, but honestly, I am starting to get lost completely

Might it not be a better idea to revert ?

**Comment by guillep** [comment]
Hi Marcus, I agree with you. We should revert. That was the rule we set.

@hernanmd it is important for us that this gets fixed fast, either by reverting or by fixing it.
It simplifies the management of all the other ~60 PRs we have.

**Comment by hernanmd** [comment]
> this adds one more failing testNoUncategorizedMethods

Hi @MarcusDenker, could you share a link to see the failed tests? 
I also got lost in the issues here.

**Comment by MarcusDenker** [comment]
The failing tests are here:

https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/job/PR-16853/8/testReport/

<blockquote><div><strong><a href="https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/job/PR-16853/8/testReport/">Test pending pull request and branch Pipeline/PR-16853 #8 Test Results [Jenkins]</a></strong></div></blockquote>

**Review by MarcusDenker — APPROVED** [review]

**Comment by MarcusDenker** [comment]
Thanks!

**Pull request merged**